lib/process-release.js

```diff
@@ -0,0 +1,129 @@
+  distBaseUrl += '/v' + version + '/'
+
+  // new style, based on process.release so we have a lot of the data we need
+  if (defaultRelease && defaultRelease.headersUrl && !overrideDistUrl) {
+    baseUrl = url.resolve(defaultRelease.headersUrl, './')
+    libUrl32 = resolveLibUrl(name, defaultRelease.libUrl || baseUrl || distBaseUrl, 'x86', version)
+    libUrl64 = resolveLibUrl(name, defaultRelease.libUrl || baseUrl || distBaseUrl, 'x64', version)
+
+    return {
+      version: version,
+      semver: versionSemver,
+      name: name,
+      baseUrl: baseUrl,
+      tarballUrl: defaultRelease.headersUrl,
+      shasumsUrl: url.resolve(baseUrl, 'SHASUMS256.txt'),
+      versionDir: (name !== 'node' ? name + '-' : '') + version,
+      libUrl32: libUrl32,
+      libUrl64: libUrl64,
+      libPath32: normalizePath(path.relative(url.parse(baseUrl).path, url.parse(libUrl32).path)),
+      libPath64: normalizePath(path.relative(url.parse(baseUrl).path, url.parse(libUrl64).path))
+    }
+  }
+
+  // older versions without process.release are captured here and we have to make
+  // a lot of assumptions, additionally if you --target=x.y.z then we can't use the
+  // current process.release
+
+  baseUrl = distBaseUrl
+  libUrl32 = resolveLibUrl(name, baseUrl, 'x86', version)
+  libUrl64 = resolveLibUrl(name, baseUrl, 'x64', version)
+  // making the bold assumption that anything with a version number >3.0.0 will
+  // have a *-headers.tar.gz file in its dist location, even some frankenstein
+  // custom version
+  tarballUrl = url.resolve(baseUrl, name + '-v' + version + (semver.satisfies(version, '>=3') ? '-headers' : '') + '.tar.gz')
+
+  return {
+    version: version,
+    semver: versionSemver,
+    name: name,
+    baseUrl: baseUrl,
+    tarballUrl: tarballUrl,
+    shasumsUrl: url.resolve(baseUrl, 'SHASUMS256.txt'),
+    versionDir: (name !== 'node' ? name + '-' : '') + version,
+    libUrl32: libUrl32,
+    libUrl64: libUrl64,
+    libPath32: normalizePath(path.relative(url.parse(baseUrl).path, url.parse(libUrl32).path)),
+    libPath64: normalizePath(path.relative(url.parse(baseUrl).path, url.parse(libUrl64).path))
+  }
+}
+
+function normalizePath (p) {
+  return path.normalize(p).replace(/\\/g, '/')
+}
+
+function resolveLibUrl (name, defaultUrl, arch, version) {
+  var base = url.resolve(defaultUrl, './')
+    , isV3 = semver.satisfies(version, '^3')
+    , hasLibUrl = bitsre.test(defaultUrl) || (isV3 && bitsreV3.test(defaultUrl))
+
+  if (!hasLibUrl) {
+    // let's assume it's a baseUrl then
+    if (semver.gte(version, '1.0.0'))
+      return url.resolve(base, 'win-' + arch  +'/' + name + '.lib')
+    // prior to io.js@1.0.0 32-bit node.lib lives in /, 64-bit lives in /x64/
+    return url.resolve(base, (arch === 'x64' ? 'x64/' : '') + name + '.lib')
+  }
+
+  // else we have a proper url to a .lib, just make sure it's the right arch
+  return defaultUrl.replace(isV3 ? bitsreV3 : bitsre, arch)
```

Comment:
This could theoretically fail if someone uses a custom URL at `mx64.company.com` or `files.company.com/arch/x86`?